backtrace-library/src/androidTest/java/backtraceio/library/breadcrumbs/BacktraceBreadcrumbsTest.java

```diff
@@ -0,0 +1,635 @@
+                assertEquals("manual", parsedBreadcrumb.get("type"));
+                assertEquals("info", parsedBreadcrumb.get("level"));
+                // Timestamp should be convertible to a number
+                Long.parseLong(parsedBreadcrumb.get("timestamp"));
+                // Id should be convertible to a number, and should skew larger (earlier breadcrumbs rolled off)
+                assertTrue(Integer.parseInt(parsedBreadcrumb.get("id")) > 45);
+            }
+
+            Comparator<Map<String, String>> mapComparator = new Comparator<Map<String, String>>() {
+                public int compare(Map<String, String> m1, Map<String, String> m2) {
+                    return Integer.parseInt(m1.get("id")) - Integer.parseInt(m2.get("id"));
+                }
+            };
+
+            Collections.sort(parsedBreadcrumbList, mapComparator);
+
+        } catch(Exception ex) {
+            fail(ex.getMessage());
+        }
+    }
+
+    @Test
+    public void breadcrumbsEnduranceTest() {
+        int numIterationsPerThread = 200;
+        int numThreads = 10;
+
+        try {
+            BacktraceBreadcrumbs backtraceBreadcrumbs = new BacktraceBreadcrumbs(context);
+            backtraceBreadcrumbs.enableBreadcrumbs();
+
+            for (int i = 0; i < numThreads; i++) {
+                new Thread(new BreadcrumbLogger(backtraceBreadcrumbs, numIterationsPerThread)).start();
+            }
+
+            List<String> breadcrumbLogFileData = readBreadcrumbLogFiles(backtraceBreadcrumbs);
+
+            List<Map<String, String>> parsedBreadcrumbList = new ArrayList<Map<String, String>>();
+            // First breadcrumb is configuration breadcrumb
+            // We start from the second breadcrumb
+            for (int i = 1; i < breadcrumbLogFileData.size(); i++)
+            {
+                Map<String, String> parsedBreadcrumb = parseBreadcrumb(breadcrumbLogFileData.get(i));
+                parsedBreadcrumbList.add(parsedBreadcrumb);
+                assertEquals("I am a breadcrumb", parsedBreadcrumb.get("message"));
+                assertNotNull(parsedBreadcrumb.get("From_Thread"));
+                assertEquals("manual", parsedBreadcrumb.get("type"));
+                assertEquals("info", parsedBreadcrumb.get("level"));
+                // Timestamp should be convertible to a number
+                Long.parseLong(parsedBreadcrumb.get("timestamp"));
+                // Id should be convertible to a number
+                Integer.parseInt(parsedBreadcrumb.get("id"));
+            }
+
+            Comparator<Map<String, String>> mapComparator = new Comparator<Map<String, String>>() {
+                public int compare(Map<String, String> m1, Map<String, String> m2) {
+                    return Integer.parseInt(m1.get("id")) - Integer.parseInt(m2.get("id"));
+                }
+            };
+
+            Collections.sort(parsedBreadcrumbList, mapComparator);
```

Comment:
Why we are sorting it if we don't care about result?

Reply:
Will remove this, thank you!